runtime/World/Entity.cpp

```diff
@@ -61,17 +62,40 @@ namespace Spartan
             for (shared_ptr<Component> component_original : entity->GetAllComponents())
             {
                 /*
-                 * TODO: Clone() requires ownership and hierarchy refactor.
-                 * TEMP: 'if' block prevents crash on Copy->Paste Entity through WorldViewer.
-                 * Mesh defaults to 'default_cube'.
+                 * TODO:
+                 * Transfer handling Mesh clone of Renderable component functionality to Renderable class. 
+                 * Currently this 'if' block prevents Mesh defaulting to Renderer::GetStandardMesh(Cube).
                  */
-                 if (component_original !=nullptr)
-                 {
+                if (component_original !=nullptr)
+                {
                     // component
                     shared_ptr<Component> component_clone = clone->AddComponent(component_original->GetType());
 
-                    // component's properties
-                    component_clone->SetAttributes(component_original->GetAttributes());
+                    if (component_original->GetType() == ComponentType::Renderable)
+                    {
+                        shared_ptr<Renderable> renderable_component = std::dynamic_pointer_cast<Renderable>(component_original);
+                        shared_ptr<Renderable> renderable_component_clone = std::dynamic_pointer_cast<Renderable>(component_clone);
+
+                        auto vc          = renderable_component->GetVertexCount();
+                        auto vo          = renderable_component->GetVertexOffset();
+                        auto gn          = renderable_component->GetGeometryName();
+                        auto gt          = renderable_component->GetGeometryType();
+                        auto aabb        = renderable_component->GetAabb();
+                        auto bb          = renderable_component->GetBoundingBox();
+                        auto ic          = renderable_component->GetIndexCount();
+                        auto io          = renderable_component->GetIndexOffset();
+                        Mesh* mesh_clone = renderable_component->GetMesh();
+
+                        // renderable component's properties
+                        renderable_component_clone->SetAttributes(renderable_component->GetAttributes());
+                        renderable_component_clone->SetGeometry(gn, io, ic, vo, vc, bb, mesh_clone);
```

Comment:
This works around the issue but doesn't fix it. You must have a look at Renderable.cpp, at the constructor.